test/unit/test_util.py

```diff
@@ -65,3 +67,50 @@ def inner(_a: Annotated[str, Inject(), Inject(param="foo")]): ...
 
 class MyCustomClass:
     pass
+
+
+def _sample_function():
+    pass
+
+
+class TestGetGlobals:
+    def test_returns_globals_for_regular_function(self):
+        # GIVEN
+        func = _sample_function
+
+        # WHEN
+        result = get_globals(func)
+
+        # THEN
+        assert result is _sample_function.__globals__
+
+    def test_returns_globals_for_class(self):
+        # GIVEN
+        cls = MyCustomClass
+
+        # WHEN
+        result = get_globals(cls)
+
+        # THEN
+        assert "_sample_function" in result
+        assert "MyCustomClass" in result
+
+    def test_unwraps_functools_partial(self):
+        # GIVEN
+        partial_func = functools.partial(_sample_function)
+
+        # WHEN
+        result = get_globals(partial_func)
+
+        # THEN
+        assert result is _sample_function.__globals__
+
+    def test_unwraps_nested_functools_partial(self):
+        # GIVEN
+        partial_func = functools.partial(functools.partial(functools.partial(_sample_function)))
+
+        # WHEN
+        result = get_globals(partial_func)
+
+        # THEN
+        assert result is _sample_function.__globals__
```

Comment:
These and the above test_returns_globals_for_regular_function are the same test. We can just parametrize them using `pytest.mark.parametrize` to give it all the different inputs we want to test it with.

Reply:
done